ext/scheduler/airflow/__lib.py

```diff
@@ -264,7 +265,7 @@ def _are_all_job_runs_successful(self, schedule_time_window_start, schedule_time
             self.log.info("job_run api response :: {}".format(api_response))
         except Exception as e:
             self.log.warning("error while fetching job runs :: {}".format(e))
-            return False
+            raise AirflowFailException(e)
```

Comment:
@deryrahman we need to retry for some errors like internal server error, we need to fail when retry is a waste like.
Can we handle it accordingly. We need to understand different error types.

Reply:
We have investigated that all the errors (non 200 status code) from the API are not worth retrying. It will only retry when there's any run status which are not success, which already handled by this [LoC](https://github.com/goto/optimus/pull/82/files#diff-9be218956b24cf39372fe215f7500dbe3737e297fcae28c25c9bb81051d462d5R270) cc: @Mryashbhardwaj